.github/workflows/build.yaml

```diff
@@ -4,27 +4,46 @@ on:
   push:
     branches: [ rolling ]
   workflow_dispatch:
+  schedule:
+    # Run every morning to detect flakiness and broken dependencies
+    - cron: '03 5 * * *'
 defaults:
   run:
     shell: bash
 jobs:
-  test:
-    runs-on: ubuntu-latest
+  build_and_test:
+    name: build_and_test_${{ matrix.BUILD_TYPE }}_${{ matrix.ROS_DISTRO }}
     strategy:
       fail-fast: false
       matrix:
-        distro: ['iron', 'jazzy', 'rolling']
+        include:
+          # Rolling (source)
+          - ROS_DISTRO: rolling
+            BUILD_TYPE: source
+            DOCKER_IMAGE: ubuntu:noble
+          # Jazzy (binary)
+          - ROS_DISTRO: jazzy
+            BUILD_TYPE: binary
+            DOCKER_IMAGE: ubuntu:noble
+          # Iron (binary)
+          - ROS_DISTRO: iron
+            BUILD_TYPE: binary
+            DOCKER_IMAGE: ubuntu:jammy
+    env:
+      ROS2_REPOS_FILE_URL: 'https://raw.githubusercontent.com/ros2/ros2/${{ matrix.ROS_DISTRO }}/ros2.repos'
+    runs-on: ubuntu-latest
     container:
-      image: ros:${{ matrix.distro }}-ros-base
-    timeout-minutes: 30
+      image: ${{ matrix.DOCKER_IMAGE }}
```

Comment:
Our CI times have increased  by 2mins by installed ROS binaries for cases where `BUILD_TYPE` is `binary`. Can we use an if condition here to use `ros:${{ matrix. ROS_DISTRO }}-ros-base` here if `matrix.BUILD_TYPE == 'binary'`; else `ubuntu:noble`. We won't require the DOCKER_IMAGE variable anymore.